bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingRegistryImpl.java

```diff
@@ -248,7 +248,13 @@ public Thing createThingOfType(ThingTypeUID thingTypeUID, ThingUID thingUID, Thi
         for (ThingHandlerFactory thingHandlerFactory : thingHandlerFactories) {
             if (thingHandlerFactory.supportsThingType(thingTypeUID)) {
                 Thing thing = thingHandlerFactory.createThing(thingTypeUID, configuration, thingUID, bridgeUID);
-                thing.setLabel(label);
+                if (thing == null) {
+                    logger.warn(
+                            "Cannot create thing of type '{}'. Binding '{}' says it supports it, but it could not be created.",
+                            thingTypeUID, thingHandlerFactory.getClass().getName());
+                } else {
+                    thing.setLabel(label);
```

Comment:
I don't think this should ever happen but if multiple bindings state to support thing type "foo", the first returns null on `createThing`, the seconds one would return a valid thing, shouldn't we call `createThing` for the second one, too?

If we move `return thing;` to the "not null" branch, we would at least give it a try...

Reply:
I've moved the return statement.